lib/pubsubstub/subscription.rb

```diff
@@ -0,0 +1,63 @@
+module Pubsubstub
+  class Subscription
+    include Logging
+
+    attr_reader :channels, :connection, :queue, :id
+
+    def initialize(channel_names, connection)
+      @id = Random.rand(2 ** 64)
+      @connection = connection
+      @channels = channel_names.map { |n| Channel.new(n) }
+      @queue = Queue.new
+    end
+
+    def stream(last_event_id)
+      info { "Connecting client ##{id} (#{channels.map(&:name).join(', ')})" }
+      fetch_scrollback(last_event_id)
+      subscribe
```

Comment:
> Would you be ok with fetching the scrollback after subscribing?

I'm OK with that. I can see other ways to do it, but not worth if we're the only one using it.

Another way other than what you suggested is we could make each connections be queues, subscribe them immediately but not released to the connection. We could then unshift scrollback, then release them.